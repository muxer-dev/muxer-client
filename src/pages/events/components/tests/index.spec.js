import React from "react";
import renderer from "react-test-renderer";
import Events from "../events";

jest.mock(
  "components/call-to-action-banner",
  () => "mock-call-to-action-banner"
);
jest.mock("components/spinner/loading", () => "mock-loading");
jest.mock("components/event", () => "mock-event");
jest.mock("components/event/error", () => "mock-error");
jest.mock("components/no-events", () => "mock-no-events");
jest.mock("components/event-separator", () => "mock-event-separator");

describe("Events", () => {
  let testContext;

  const createComponent = props => <Events {...props} />;

  beforeEach(() => {
    testContext = {};
    testContext.defaultProps = {
      fetchData: () => jest.fn(),
      setLocation: () => jest.fn(),
      location: "belfast",
      url: "http://localhost/",
      params: { page: 1, location: "belfast"},
      isLoading: false,
      hasErrors: false,
      hasMoreItems: true,
      upcomingEvents: [{ id: "1" }, { id: "2" }],
      recentEvents: [{ id: "1" }, { id: "2" }]
    };
  });

  describe("when the component is rendered", () => {
    beforeEach(() => {
      testContext.component = renderer.create(
        createComponent(testContext.defaultProps)
      );
    });

    it("renders correctly", () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });

  describe("when the component is rendered with errors", () => {
    beforeEach(() => {
      testContext.props = {
        ...testContext.defaultProps,
        hasErrors: true
      };

      testContext.component = renderer.create(
        createComponent(testContext.props)
      );
    });

    it("renders correctly", () => {
      expect(testContext.component).toMatchSnapshot();
    });
  });
});
