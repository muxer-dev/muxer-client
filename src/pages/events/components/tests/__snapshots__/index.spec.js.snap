// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events when the component is rendered renders correctly 1`] = `
Array [
  <div
    className="page"
>
    <mock-call-to-action-banner />
    <div>
        <div
            className="recent-events"
        >
            <mock-event-separator
                content="Recent Events"
                id="recent-events"
            />
            <mock-event
                className="recent-event"
                content={
                    Object {
                        "id": "1",
                      }
                }
            />
            <mock-event
                className="recent-event"
                content={
                    Object {
                        "id": "2",
                      }
                }
            />
        </div>
        <div
            className="events-today"
        >
            <mock-event-separator
                content="Today"
                id="today"
            />
            <mock-event
                content={
                    Object {
                        "id": "1",
                      }
                }
            />
            <mock-event
                content={
                    Object {
                        "id": "2",
                      }
                }
            />
        </div>
        <div
            className="events-in-next-seven-days"
        >
            <mock-event-separator
                content="Next 7 days"
                id="in-next-seven-days"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
        <div
            className="events-in-next-thirty-days"
        >
            <mock-event-separator
                content="Next 30 days"
                id="in-next-thirty-days"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
        <div
            className="events-in-next-three-months"
        >
            <mock-event-separator
                content="Next 3 months"
                id="in-next-three-months"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
        <div
            className="events-in-the-future"
        >
            <mock-event-separator
                content="And beyond..."
                id="in-the-future"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
    </div>
</div>,
  <span
    style={
        Object {
            "fontSize": 0,
          }
    }
/>,
]
`;

exports[`Events when the component is rendered with errors renders correctly 1`] = `
Array [
  <div
    className="page"
>
    <mock-call-to-action-banner />
    <div>
        <div
            className="recent-events"
        >
            <mock-event-separator
                content="Recent Events"
                id="recent-events"
            />
            <mock-event
                className="recent-event"
                content={
                    Object {
                        "id": "1",
                      }
                }
            />
            <mock-event
                className="recent-event"
                content={
                    Object {
                        "id": "2",
                      }
                }
            />
        </div>
        <div
            className="events-today"
        >
            <mock-event-separator
                content="Today"
                id="today"
            />
            <mock-event
                content={
                    Object {
                        "id": "1",
                      }
                }
            />
            <mock-event
                content={
                    Object {
                        "id": "2",
                      }
                }
            />
        </div>
        <div
            className="events-in-next-seven-days"
        >
            <mock-event-separator
                content="Next 7 days"
                id="in-next-seven-days"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
        <div
            className="events-in-next-thirty-days"
        >
            <mock-event-separator
                content="Next 30 days"
                id="in-next-thirty-days"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
        <div
            className="events-in-next-three-months"
        >
            <mock-event-separator
                content="Next 3 months"
                id="in-next-three-months"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
        <div
            className="events-in-the-future"
        >
            <mock-event-separator
                content="And beyond..."
                id="in-the-future"
            />
            <mock-no-events
                location="belfast"
            />
        </div>
    </div>
    <mock-error />
</div>,
  <span
    style={
        Object {
            "fontSize": 0,
          }
    }
/>,
]
`;
